src/Arcus.WebApi.Security/Authentication/Certificates/CertificateAuthenticationValidator.cs

```diff
@@ -42,8 +39,14 @@ public CertificateAuthenticationValidator(CertificateAuthenticationConfig certif
         /// <exception cref="ArgumentNullException">Thrown when the <paramref name="services"/> is <c>null</c>.</exception>
         internal async Task<bool> IsCertificateAllowedAsync(X509Certificate2 clientCertificate, IServiceProvider services)
         {
-            Guard.NotNull(clientCertificate, nameof(clientCertificate), "Certificate authentication validation requires a client certificate");
-            Guard.NotNull(services, nameof(services), "Certificate authentication validation requires a service object to retrieve registered services");
+            if (clientCertificate is null)
+            {
+                throw new ArgumentNullException(nameof(clientCertificate), "Certificate authentication validation requires a client certificate");
+            }
+            if (services is null)
+            {
```

Comment:
```suggestion
            }

            if (services is null)
            {
```
It's a bit a Microsoft style that there is a blank space between curly braces and the next line.